crypto/crypto-derive/src/lib.rs

```diff
@@ -227,3 +247,58 @@ pub fn derive_enum_signature(input: TokenStream) -> TokenStream {
         }
     }
 }
+
+#[proc_macro_derive(CryptoHasher, attributes(CryptoHasherSalt))]
```

Comment:
I should have taken a closer look at this before it was merged, I'm sorry I'm just getting around to it now.

Its pretty odd that we're `deriving` a trait not on the type that is being generated and not deriving the trait on the struct the derive macro is placed above. I would have though it would make more sense to derive the CryptoHash trait and remove the CryptoHasher trait entirely.

Reply:
Unfortunately since the CryptoHash trait is custom (= non-dérivable) for a number of structures, that specific fix is impossible. I agree there are relevant architectural fixes to be done, but I think they are out of scope for this PR and they are better addressed as a later PR that maintains the automatic salt derivation